docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -0,0 +1,428 @@
+   ToasterKit's compilation.
+   This ensures when an app is built for deployment targets prior to the symbols' move,
+   the app will look for these symbols in ToasterKit instead of ToasterKitCore.
+
+More generally, mutliple availabilities can be specified, like so:
+
+```swift
+@available(toasterOS 42, bowlOS 54, mugOS 54, *)
+@_originallyDefinedIn(module: "ToasterKit", toasterOS 57, bowlOS 69, mugOS 69)
+enum Toast { ... }
+```
+
+## `@_private(sourceFile: "FileName.swift")`
+
+Fully bypasses access control, allowing access to private declarations
+in the imported module. The imported module needs to be compiled with
+`-Xfrontend -enable-private-imports` for this to work.
+
+## `@_semantics("uniquely.recognized.id")`
+
+Allows the optimizer to make use of some key invariants in performance critical
+data types, especially `Array`. Since the implementation of these data types
+is written in Swift using unsafe APIs, without these attributes the optimizer
+would need to make conservative assumptions.
+
+Changing the implementation in a way that violates the optimizer's assumptions
+about the semantics results in undefined behavior.
+
+## `@_show_in_interface`
+
+## `@_silgen_name("cName")`
+
+Changes the symbol name for a function, similar to an ASM label in C,
+except that the platform symbol mangling (leading underscore on Darwin)
+is maintained.
+
+Since this has label-like behavior, it may not correspond to any declaration;
+if so, it is assumed that the function is implemented in C.
+
+A function defined by `@_silgen_name` is assumed to use the Swift ABI.
+
+## `@_specialize(...)`
+
+Forces generation of a specialized implementation for a generic declaration.
+
+See [Generics.rst](/docs/Generics.rst) for more details.
+
+## `@_specializeExtension`
+
+## `@_spi(spiName)`
+
+Marks a declaration as SPI (System Programming Interface), instead of API.
+Modules exposing SPI and using library evolution generate an additional
+`.private.swiftinterface` file (with `-emit-private-module-interface-path`)
+in addition to the usual `.swiftinterface` file. This private interface exposes
+both API and SPI.
+
+Clients can access SPI by marking the import as `@_spi(spiName) import Module`.
+This design makes it easy to find out which clients are using certain SPIs by
+doing a textual search.
+
+## `@_staticInitializeObjCMetadata`
+
+## `@_transparent`
+
+Marks a function to be "macro-like", i.e., it is guaranteed to be inlined
+in debug builds.
+
+See [TransparentAttr.md](/docs/TransparentAttr.md) for more details.
+
+## `@_typeEraser`
+
+## `@_weakLinked`
+
+## `@_unsafeMainActor`, `@_unsafeSendable`
+
+Marks a parameter's (function) type as `@MainActor` (`@Sendable) in Swift 6 and
+within Swift 5 code that has adopted concurrency, but non-`@MainActor`
+(non-`@Sendable) everywhere else.
+
+See the [forum post on Concurrency in Swift 5 and 6](https://forums.swift.org/t/concurrency-in-swift-5-and-6/49337)
```

Comment:
```suggestion
See the forum post [Concurrency in Swift 5 and 6](https://forums.swift.org/t/concurrency-in-swift-5-and-6/49337)
```

Reply:
Same as above, longer link text to provide better context to screen readers.